src/folio_migration_tools/marc_rules_transformation/rules_mapper_bibs.py

```diff
@@ -460,66 +461,76 @@ def handle_hrid(self, folio_instance, marc_record: Record, legacy_ids) -> None:
-                    self.migration_report.add(Blurbs.HridHandling, s)
-                    Helper.log_data_issue(legacy_ids, s, marc_record["001"])
-                else:
-                    self.migration_report.add(
-                        Blurbs.HridHandling, "Legacy bib records without 001"
-                    )
-            marc_record.add_ordered_field(new_001)
-            self.migration_report.add(Blurbs.HridHandling, "Created HRID using default settings")
-            self.instance_hrid_counter += 1
-        elif self.hrid_handling == HridHandling.preserve001:
-            value = marc_record["001"].value()
-            if value in self.unique_001s:
+            if self.task_configuration.deactivate035_from001:
                 self.migration_report.add(
-                    Blurbs.HridHandling, "Duplicate 001. Creating HRID instead"
+                    Blurbs.HridHandling, "035 generation from 001 turned off"
                 )
-                Helper.log_data_issue(
-                    legacy_ids,
-                    "Duplicate 001 for record. HRID created for record",
-                    value,
+            else:
+                str_035 = f"({f_003}){f_001}" if f_003 else f"{f_001}"
+                new_035 = Field(
+                    tag="035",
+                    indicators=[" ", " "],
+                    subfields=["a", str_035],
                 )
-                num_part = str(self.instance_hrid_counter).zfill(11)
-                folio_instance["hrid"] = f"{self.instance_hrid_prefix}{num_part}"
-                new_001 = Field(tag="001", data=folio_instance["hrid"])
-                marc_record.add_ordered_field(new_001)
-                self.instance_hrid_counter += 1
+                marc_record.add_ordered_field(new_035)
+                self.migration_report.add(Blurbs.HridHandling, "Added 035 from 001")
+            marc_record.remove_fields("001")
+
+        except Exception:
+            if "001" in marc_record:
+                s = "Failed to create 035 from 001"
+                self.migration_report.add(Blurbs.HridHandling, s)
+                Helper.log_data_issue(legacy_ids, s, marc_record["001"])
             else:
-                self.unique_001s.add(value)
-                folio_instance["hrid"] = value
-                self.migration_report.add(Blurbs.HridHandling, "Took HRID from 001")
+                self.migration_report.add(Blurbs.HridHandling, "Legacy bib records without 001")
+        marc_record.add_ordered_field(new_001)
+        self.migration_report.add(Blurbs.HridHandling, "Created HRID using default settings")
+        self.instance_hrid_counter += 1
+
+    def generate_numeric_part(self):
+        return (
+            str(self.instance_hrid_counter).zfill(11)
+            if self.common_retain_leading_zeroes
+            else str(self.instance_hrid_counter)
+        )
+
+    def preserve_001_as_hrid(self, folio_instance, marc_record, legacy_ids):
```

Comment:
Should we be copying the 001 + 003 when there is a duplicate in this case?

Reply:
Very good point. The idealistic part of me would like to have the library to fix this, but reality is that there will always be records that are not fixed. 

I made this change.